crayfish/animation.py

```diff
@@ -101,32 +101,37 @@ def prep_comp(cfg, mr, time):
     w,h = mr.outputSize().width(), mr.outputSize().height()
     dpi = mr.outputDpi()
     c = QgsComposition(mr)
+    #suppose to be always set below
+    dataset = None 
+    if cfg['layer'] and cfg['layer'].currentDataSet():
+        dataset = cfg['layer'].currentDataSet()
     if layoutcfg['type'] == 'file':
-        prepare_composition_from_template(c, cfg['layout']['file'], time)
+        prepare_composition_from_template(c, cfg['layout']['file'], time, dataset)
     else:  # type == 'default'
-        cm = prepare_composition(c, w, h, dpi, time, layoutcfg)
+        cm = prepare_composition(c, w, h, dpi, time, layoutcfg, ds=dataset)
     return c
 
 
 def composition_set_time(c, time, frmt=0):
-    timeItem = c.getComposerItemById("time")
-    if timeItem is not None:
-        if frmt == 0: # hh:mm:ss
-            txt = time_to_string(time)
-        else:  # hh.hhh
-            txt = "%06.3f" % time
-        timeItem.setText(txt)
+    for i in c.items():
```

Comment:
this function should just find time element and assign string to it. Time formatting should be done outside of the function. Also I thought we want always to use time format based on user preference, so I do not understand why we need here time_to_string function? Maybe that function can be deleted altogether to not confuse developers for the future?

Reply:
It can be refactor, but no functionality has been changed after this. The problem was with getting element ```timeItem = c.getComposerItemById("time")``` which returns generic item instead of label item.